openlibrary/plugins/openlibrary/js/partner_ol_lib.js

```diff
@@ -0,0 +1,66 @@
+/**
+ * @param {string} container
+ */
+function getIsbnToElementMap(container) {
+    const reISBN = /((978)?[0-9][0-9]{10}[0-9xX])|((978)?[0-9]{9}[0-9Xx])/;
+    const elements = Array.from(document.querySelectorAll(container));
+    const isbnElementMap = {};
+    elements.forEach((e) => {
+        const isbnMatches = e.innerHTML.match(reISBN);
+        if (isbnMatches) {
+            isbnElementMap[isbnMatches[0]] = e;
+        }
+    })
+    return isbnElementMap;
+}
+
+/**
+ * @param {string[]} isbnList
+ * @returns {Promise<Array>}
+ */
+async function getAvailabilityDataFromArchiveOrg(isbnList) {
+    const apiBaseUrl = 'https://archive.org/services/availability'
+    const apiUrl = `${apiBaseUrl}?isbn=${isbnList.join(',')}`;
+    const response = await fetch(apiUrl);
+    const jsonResponse = await response.json();
+    return jsonResponse.responses;
+}
+
+/**
+ * @param {object} options
+ * @param {string} options.bookContainer class name of the HTML element associated with a book. We will try to find a book identifier (eg ISBN) in each of these.
+ * @param {string} options.selectorToPlaceBtnIn The class name of the HTML element that we will add the Open Library button to. Each `bookContainer` should have one of these.
+ * @param {string} [options.textOnBtn] The text on the button
+ *
+ * @example
+ * addOpenLibraryButtons({
+ *    bookContainer: ".book-container",
+ *    selectorToPlaceBtnIn: ".btn-container",
+ *    textOnBtn: "Open Library!"
+ * });
+ */
+async function addOpenLibraryButtons(options) {
+    const {bookContainer, selectorToPlaceBtnIn, textOnBtn} = options
+    if (bookContainer === undefined || selectorToPlaceBtnIn === undefined) {
```

Comment:
Let's make the default be the bookContainer
```suggestion
    if (bookContainer === undefined) {
```